hack/install_kustomize.sh

```diff
@@ -123,31 +134,38 @@ s390x)
     ;;
 esac
 
-releases=$(curl -s $release_url)
+releases=$(curl -s "$release_url")
 
 if [[ $releases == *"API rate limit exceeded"* ]]; then
   echo "Github rate-limiter failed the request. Either authenticate or wait a couple of minutes."
   exit 1
 fi
 
-RELEASE_URL=$(echo "${releases}" |\
-  grep browser_download.*${opsys}_${arch} |\
-  cut -d '"' -f 4 |\
-  sort -V | tail -n 1)
+RELEASE_URL="$(find_release_url "$releases" "$opsys" "$arch")"
+
+if [[ "$arch" == "arm64" ]] && [[ -z "$RELEASE_URL" ]] ; then
+    # fallback to the old behavior of downloading amd64 binaries on aarch64 systems.
+    # People might have qemu-binfmt-misc installed, so it worked for them previously.
+    echo "Version $version does not exist for ${opsys}_arm64, trying ${opsys}_amd64 instead."
+    RELEASE_URL="$(find_release_url "$releases" "$opsys" "amd64")"
+    if [[ -n "$RELEASE_URL" ]]; then
+      arch=amd64
```

Comment:
Since we've informed the user that we're about to try amd64, I think we should change this unconditionally before L150 and then not need to hard code it there. Otherwise the messaging is a little confusing:

```shell
# attempting to install v3.2.0 on an m1 mac
Version v3.2.0 does not exist for darwin_arm64, trying darwin_amd64 instead.
Version v3.2.0 does not exist or is not available for darwin/arm64. # so, did we actually try amd64 or not?
```

Reply:
done :)